python3Packages.pygments: 2.13.0 -> 2.14.0

ncfavier opened the pull request:
https://github.com/pygments/pygments/releases/tag/2.14.0

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ncfavier committed:
python3Packages.pygments: 2.13.0 -> 2.14.0

https://github.com/pygments/pygments/releases/tag/2.14.0

ncfavier commented:
Anything else I should test beyond basic functionality of pygments itself?

Ofborg seems useless here...

SuperSandro2000 commented:
@ofbrog eval

SuperSandro2000 commented:
> Anything else I should test beyond basic functionality of pygments itself?

We are running tests, so should be fine.

ofborg fails because cache is missing.

ncfavier commented:
No dice. `python3Packages.pygments.tests` builds fine on the PR's tip (not rebased on staging).

SuperSandro2000 commented:
It doesn't build at all on rebased on staging because sphinx tests fail right now.

ncfavier commented:
Where do we go from here? Surely I can't be expected to rebuild the world on my laptop just to update a package that runs fine on master?

SuperSandro2000 commented:
We wait until that is fixed :P

@ofborg eval

SuperSandro2000 commented:
I think the failure is caused by the pygments update. I couldn't produce it before.

ncfavier commented:
https://github.com/sphinx-doc/sphinx/commit/965768bfda2a00ba6466cdb12a7a46efdce47023

Why is our sphinx so outdated?

ncfavier commented:
https://github.com/NixOS/nixpkgs/pull/210338

ncfavier committed:
sphinx: disable tests that fail with pygments 2.14

https://github.com/sphinx-doc/sphinx/commit/965768bfda2a00ba6466cdb12a7a46efdce47023

ncfavier commented:
Since the sphinx update might take time, I've just disabled the failing tests.

ncfavier commented:
Is this good enough to merge?

SuperSandro2000 commented on pkgs/development/python-modules/sphinx/default.nix:
```diff
@@ -106,6 +106,12 @@ buildPythonPackage rec {
+
+    # fails with pygments 2.14
+    # TODO remove for sphinx 6
+    "test_viewcode"
+    "test_additional_targets_should_be_translated"
+    "test_additional_targets_should_not_be_translated"
```
Does this break the actual code? I am a bit reluctant.

ncfavier replied:
Judging by https://github.com/sphinx-doc/sphinx/commit/965768bfda2a00ba6466cdb12a7a46efdce47023 it seems like minor formatting differences.

SuperSandro2000 reviewed (approved):
@ofborg build python310Packages.pygments python310Packages.sphinx

SuperSandro2000 committed:
python310Packages.rich: 13.0.0 -> 13.3.0

Fixes pygments compatibility
https://github.com/Textualize/rich/blob/HEAD/CHANGELOG.md

SuperSandro2000 commented:
I just pushed an update for rich to this branch because the latest update had a fix for pygments.

ncfavier commented:
The rich update introduces a circular dependency (for python 3.11 only):

```
rich → markdown-it-py
     → pytest-regressions (tests)
     → pandas
     → hypothesis (tests)
     → sphinx-codeautolink (doc)
     → ipython
     → stack-data
     → executing
     → rich (tests, python >= 3.11)
```

ncfavier committed:
python3Packages.pandas: break dependency cycle with hypothesis docs

hypothesis → sphinx-autocodelink → ipython → stack-data → executing →
rich → markdown-it-py → pytest-regressions → pandas

ncfavier commented:
Broke the cycle at `hypothesis` since there seems to be precedent for this.

ncfavier commented:
rich 13.3.0 breaks textual...

ncfavier commented:
This is still broken: now ipython's tests are failing.

mweinelt commented:
Step by step.

ncfavier commented:
Okay, https://github.com/NixOS/nixpkgs/pull/212441 should fix ipython via https://github.com/ipython/ipython/pull/13882.

SuperSandro2000 commented on pkgs/development/python-modules/textual/default.nix:
```diff
@@ -48,6 +48,8 @@ buildPythonPackage rec {
     typing-extensions
   ];
 
+  doCheck = false; # broken with rich 13.3.0
```
Please also add why

ncfavier replied:
I've updated textual instead (which required a markdown-it-py update), now tests pass.

SuperSandro2000 commented on pkgs/development/python-modules/pandas/default.nix:
```diff
@@ -49,7 +49,8 @@ buildPythonPackage rec {
   nativeCheckInputs = [
     glibcLocales
-    hypothesis
+    # hypothesis indirectly depends on pandas to build its documentation
+    (hypothesis.override { enableDocumentation = false; })
```
Not sure if we should move this to python-packages

ncfavier replied:
Presumably we still want the top-level `hypothesis` to have documentation?

SuperSandro2000 replied:
I mean if we should move the overwrite to the attr set passed to pandas callPackage.

ncfavier committed:
python3Packages.markdown-it-py: 2.1.0 -> 2.2.0

https://github.com/executablebooks/markdown-it-py/releases/tag/v2.2.0

ncfavier committed:
python3Packages.textual: 0.10.1 -> 0.11.1

ncfavier commented:
`pygments`, `pygments.tests` build fine on master with https://github.com/NixOS/nixpkgs/pull/215904 applied. Maybe it's time to merge both PRs?

SuperSandro2000 commented:
It is probably a better idea to wait for a python-updates run to fix possible other regressions in big style.

mweinelt commented:
Meh. wrong destination branch hit.
